### PR TITLE
Paper/afrcnn

### PR DESCRIPTION
**Pull request by czotti**

### CODE REVIEW

File: deep-learning/_posts/2017-08-04-A-Fast-RCNN.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+     <img src="/deep-learning/images/afrcnn/network.png"/>
+</div>
+
+## Training
+
+They use a pre-trained Fast-RCNN network on ImageNet. The training is done in a staged manner. First, they train FRCNN on the new database for localization. After, they fix the weights of the detector and train the adversarial model to create occlusion or rotation. In the end, they train both models at the same time to improve the detector network.
+
+## Results
+
+They report result three dataset, [PASCAL VOC 2007](http://host.robots.ox.ac.uk/pascal/VOC/voc2007/), [PASCAL VOC 2012](http://host.robots.ox.ac.uk/pascal/VOC/), and [MSCOCO](http://cocodataset.org).
```
Comment (Dref360) [inline deep-learning/_posts/2017-08-04-A-Fast-RCNN.md]: result on three datasets :*

File: deep-learning/_posts/2017-08-04-A-Fast-RCNN.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+Paper :
+   - [code](https://github.com/xiaolonw/adversarial-frcnn)
+
+This paper shows a way to train an adversarial network to generate occlusion/deformation changes to help an object detector network to generalize better.
+
+As an object detector network they use [Faster-RCNN]({% post_url /deep-learning/2017-05-14-fastrcnn %}).
+
+## Model
+
+The adversarial model has two parts (Fig.). The first one is a spatial dropout occlusion (ASDN), it generates occlusion maps on the object features after the ROI layer.
```
Comment (ppoulin91) [inline deep-learning/_posts/2017-08-04-A-Fast-RCNN.md]: Fig. ?

File: deep-learning/_posts/2017-08-04-A-Fast-RCNN.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+To select which pixel to mask, they use importance sampling on the mask and keep only the top $$ \frac{1}{3} $$ pixels.
+
+The second part is a spatial transformer network (ASTN), it generates deformation on the object features. The deformation is mostly focused on the rotation. They found that limiting the rotation is a key factor in their experiment to not have upside down rotations and fool the prediction network. The rotation limit is in between $$ -10^\circ $$ to $$ 10^\circ $$. The feature maps are divided into 4 blocks given the channel axis and different rotations are applied on these blocks.
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-08-04-A-Fast-RCNN.md]: > divided into 4 blocks given the channel axis

I would replace "given" by "along"

This clarifies things! During your presentation I thought it was divided spatially!

Comment (czotti) [reply deep-learning/_posts/2017-08-04-A-Fast-RCNN.md]: Yes, I spell it wrong during the presentation.

**Pull request merged**